### PR TITLE
Fix B2B mode

### PR DESCRIPTION
**Pull request by maximebiloe**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Fix B2B mode. Now display Identification number, company name and VAT number fields when registering if the B2B mode is enabled and add a hook to let VAT number module check the field |
| Type? | bug fix |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-828 |

### CODE REVIEW

**Comment by prestamodule** [comment]
Address::validateController() is now useless as it's identical to ObjectModel::validateController(), you should remove it ;)

**Comment by maximebiloe** [comment]
@prestamodule you're right. I've removed the function.

File: classes/form/CustomerAddressForm.php [hunk 1]
```diff
@@ -86,6 +86,10 @@ public function validate()
             }
         }
 
+        if ($hookReturn = Hook::exec('actionValidateCustomerAddressForm', array('form' => $this)) != '') {
+            $is_valid &= $hookReturn;
```
Comment (Quetzacoalt91) [inline classes/form/CustomerAddressForm.php]: Maybe you should handle unexpected type of value by adding a (bool) cast before `$hookReturn` ?

Comment (maximebiloe) [inline classes/form/CustomerAddressForm.php]: Good point! Just added the cast.

**Pull request merged**